src/tests/string/test_string_operator.f90

```diff
@@ -0,0 +1,123 @@
+! SPDX-Identifer: MIT
+module test_string_operator
+    use stdlib_error, only : check
+    use stdlib_string_type, only : string_type, assignment(=), len, &
+        operator(.gt.), operator(.lt.), operator(.ge.), operator(.le.), &
+        operator(.ne.), operator(.eq.), operator(//)
+    implicit none
+
+contains
+
+    subroutine test_gt
+        type(string_type) :: string
+        logical :: res
+
+        string = "bcd"
+        res = string > "abc"
+        call check(res .eqv. .true.)
```

Comment:
I'm curious, why `call check(res .eqv. .true.)` instead of `call check(res)`? And similar for `call check(res .eqv. .false.)` instead of `call check(.not. res)`. Is there any practical difference or is it just style?

Reply:
Yes that could be simplified, I created those tests from the examples in the specs where I added a comment after each operation to show the current state / content of the variable and batch translated them with vim into `check` functions.

Reply:
I also suggest to further isolate each test. As they are now, these tests also depend on the assignment operator (`string = "bcd"`). So if there was a regression that broke the assignment, leading to, say, `string` remaining empty after `string = "bcd", the comparison test would fail, but it wouldn't be the fault of the comparison. So for these tests I suggest a more concise:

```
call check(string_type("bcd") > "abc")
```

and similar for others.

Reply:
Which in turn would dependent on the custom constructor instead of the assignment. The only safe way would be to test both string types created by assignment and custom constructor.

Reply:
I decided to leave the current tests in as usage examples and have a little bit of Fortran procedure pointer magic generate tests for broader functionality coverage.